app/controllers/PSASchemeDetailsController.scala

```diff
@@ -34,18 +39,37 @@ class PSASchemeDetailsController @Inject()(appConfig: FrontendAppConfig,
                                            schemeDetailsConnector: SchemeDetailsConnector,
                                            schemeTransformer: SchemeDetailsMasterSection,
                                            authenticate: AuthAction,
-                                           errorHandler: ErrorHandler
-                                       )(implicit val ec: ExecutionContext) extends FrontendController with I18nSupport {
+                                           errorHandler: ErrorHandler,
+                                           featureSwitchManagementService: FeatureSwitchManagementService,
+                                           @SchemeDetailsReadOnly schemeDetailsReadOnlyCacheConnector: UserAnswersCacheConnector
+                                          )(implicit val ec: ExecutionContext) extends FrontendController with I18nSupport {
 
   def onPageLoad(srn: String): Action[AnyContent] = authenticate.async {
     implicit request =>
-      schemeDetailsConnector.getSchemeDetails(request.psaId.id, schemeIdType ="srn", srn).flatMap { scheme =>
-        if (scheme.psaDetails.exists(_.id == request.psaId.id)) {
-          val schemeDetailMasterSection = schemeTransformer.transformMasterSection(scheme)
-          Future.successful(Ok(psa_scheme_details(appConfig, schemeDetailMasterSection, scheme.schemeDetails.name, srn)))
-        } else {
-          Future.successful(NotFound(errorHandler.notFoundTemplate))
-        }
+      if (featureSwitchManagementService.get("is-variations-enabled")) {
```

Comment:
do we have flag defined in Toggles file? Can we use it from that file instead of hard coding

Reply:
Done